diff --git a/librecomp/CMakeLists.txt b/librecomp/CMakeLists.txt
index fcc3337..2ebc47f 100644
--- a/librecomp/CMakeLists.txt
+++ b/librecomp/CMakeLists.txt
@@ -48,11 +48,13 @@ target_include_directories(librecomp PRIVATE
     "${CMAKE_CURRENT_SOURCE_DIR}/include/librecomp"
 )
 
-target_compile_options(librecomp PRIVATE
-#    -Wall
-#    -Wextra
-    -Wno-unused-parameter
-)
+if(NOT MSVC)
+    target_compile_options(librecomp PRIVATE
+    #    -Wall
+    #    -Wextra
+        -Wno-unused-parameter
+    )
+endif()
 
 if (WIN32)
     add_compile_definitions(NOMINMAX)
diff --git a/ultramodern/CMakeLists.txt b/ultramodern/CMakeLists.txt
index 5593172..3c3e1b6 100644
--- a/ultramodern/CMakeLists.txt
+++ b/ultramodern/CMakeLists.txt
@@ -30,11 +30,13 @@ target_include_directories(ultramodern PUBLIC
     "${PROJECT_SOURCE_DIR}/../thirdparty/sse2neon"
 )
 
-target_compile_options(ultramodern PRIVATE
-#    -Wall
-#    -Wextra
-    -Wno-unused-parameter
-)
+if(NOT MSVC)
+    target_compile_options(ultramodern PRIVATE
+    #    -Wall
+    #    -Wextra
+        -Wno-unused-parameter
+    )
+endif()
 
 if (WIN32)
     add_compile_definitions(NOMINMAX)
diff --git a/ultramodern/src/events.cpp b/ultramodern/src/events.cpp
index 518529d..265d246 100644
--- a/ultramodern/src/events.cpp
+++ b/ultramodern/src/events.cpp
@@ -68,6 +68,7 @@ static struct {
         void update_vi() {
             ViState* next_state = get_next_state();
             const OSViMode* next_mode = next_state->mode;
+            if (next_mode == nullptr) return; // Guard against uninitialized VI state
             const OSViCommonRegs* common_regs = &next_mode->comRegs;
             const OSViFieldRegs* field_regs = &next_mode->fldRegs[field];
             PTR(void) framebuffer = osVirtualToPhysical(next_state->framebuffer);
